src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/PlatformDetection.cs

```diff
@@ -204,6 +204,38 @@ private static bool GetSsl3Support()
             return (IsOSX || (IsLinux && OpenSslVersion < new Version(1, 0, 2) && !IsDebian));
         }
 
+        private static bool GetTls13Support()
+        {
+            if (IsWindows)
+            {
+                if (!IsWindows10Version2004OrGreater)
+                {
+                    return false;
+                }
+
+                string clientKey = @"HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Client";
+                string serverKey = @"HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3\Server";
+
+                object client, server;
+                try
+                {
+                    client = Registry.GetValue(clientKey, "Enabled", null);
+                    server = Registry.GetValue(serverKey, "Enabled", null);
+                    if (client is int c && server is int s)
+                    {
+                        return c == 1 && s == 1;
+                    }
+                }
+                catch { };
+                // assume no if key is missing or on error.
+                return false;
+            }
+            else
+            {
+                return !IsOSX && (OpenSslVersion.CompareTo(new Version(1,1,1)) >= 0);
```

Comment:
Might be a little bit more understandable as:
```suggestion
                return !IsOSX && OpenSslVersion >= new Version(1,1,1);
```